php/php.editor/src/org/netbeans/modules/php/editor/model/VariableScopeFinder.java

```diff
@@ -45,10 +51,22 @@ public VariableScope find(final Scope scope, final int offset, final ScopeRangeA
     }
 
     public VariableScope find(final List<? extends ModelElement> elements, final int offset, final ScopeRangeAcceptor scopeRangeAcceptor) {
-        return findWrapper(elements, offset, scopeRangeAcceptor).getVariableScope();
+        AtomicBoolean isLazilyScanned = new AtomicBoolean(false);
+        VariableScope variableScope = findWrapper(elements, offset, scopeRangeAcceptor, isLazilyScanned).getVariableScope();
+        if (isLazilyScanned.get()) {
+            // some scopes may be added new elements when LazyBuild elements are scanned.
+            // so, find again.
+            // e.g. Source instances are cached as weak references.
+            // so, ParserResult may not be the same instance even if the FileObject is the same.
+            // it means that new Model and new ModelVisitor are created again. in such case, LazyBuild elements may not be scanned yet.
+            LOGGER.log(Level.FINE, "(LazyBuild)Scope is scanned."); // NOI18N
+            isLazilyScanned.set(false);
```

Comment:
Maybe better [`compareAndSet()`](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/AtomicBoolean.html#compareAndSet-boolean-boolean-)?

Reply:
Ha, it is running in just one thread, right? In such a case, not needed, of course.

Reply:
yes, will remove.

Reply:
Oops, will change it to your suggestion :)

Reply:
If it is running in only one thread (as here I guess, in the _parsing_ thread) then `AtomicBoolean` is just a holder for a boolean. So, using `compareAndSet()` is IMHO not necessary.

Reply:
I seem to misunderstand something. Sorry... I'll revert it if I modify this class in the future. Thank you for your review :)

Reply:
No problem, my fault because of my misleading/confusing comment.